docs/rules/no-untyped-mock-factory.md

```diff
@@ -0,0 +1,57 @@
+# Disallow using `jest.mock()` factories without an explicit type parameter (`no-untyped-mock-factory`)
+
+<!-- end auto-generated rule header -->
+
+By default, `jest.mock` and `jest.doMock` allow any type to be returned by a
+mock factory. A generic type parameter can be used to enforce that the factory
+returns an object with the same shape as the original module, or some other
+strict type. Requiring a type makes it easier to use TypeScript to catch changes
+needed in test mocks when the source module changes.
+
+## Rule details
```

Comment:
Let's add a warning about the fact that this rule should only be run on TS files, and that overrides can be used to help with that:

```suggestion
needed in test mocks when the source module changes.

> **Warning**
> 
> This rule expects to be run on TypeScript files **only**. If you are using a codebase that has a mix of JavaScript and TypeScript tests, you can use [overrides](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#how-do-overrides-work) to apply this rule to just your TypeScript test files.

## Rule details
```

(feel free to make changes to what I've written)